src/server/http/__tests__/xsrf.js

```diff
@@ -29,10 +30,23 @@ describe('xsrf request filter', function () {
 
     await kbnServer.ready();
 
-    const routeMethods = nonDestructiveMethods.filter(method => method !== 'HEAD').concat(destructiveMethods);
     kbnServer.server.route({
       path: '/xsrf/test/route',
-      method: routeMethods,
+      method: nonDestructiveMethods.filter(method => method !== 'HEAD'),
+      handler: function (req, reply) {
+        reply(null, 'ok');
+      }
+    });
+
+    kbnServer.server.route({
+      path: '/xsrf/test/route',
+      method: destructiveMethods,
+      config: {
+        // Disable payload parsing to make HapiJS server accept any content-type header.
```

Comment:
What happens in our "production setup" vs "test setup" here? If we special-case the config here, but it isn't specified like this in our actual routes, will it work as expected?

Maybe we should add 1 or 2 integration tests that checks for the 400 response at least? See `test/api_integration/apis` in https://github.com/elastic/kibana/pull/11114/files for some examples.